### PR TITLE
Test runner utility

### PR DESCRIPTION
**Pull request by DavidBM**
After experimenting a bit I found a way to keep the test directories clean, even if the old one died.

Basically, it creates a random named directory inside of tests and clean it up after the test, regardless of if it fails or succeeds. 

The only limitation I found is that there are not yet async lambdas in rust, only code blocks. Therefore, I cannot automate the async stuff. You can check the test we have to see how it looks.

### CODE REVIEW

**Comment by DavidBM** [comment]
Well... That took quite a long time trying and I still don't know how I feel about this. Look at the last commit 8a1bf9c and check how the test is created now. 

Basically, in each test body we need to 

```rust
    let runner = build_basic_runner();

    runner.run(move |folder_name| Box::new(async move {

    }));
```

This is a hack until `impl Future` can be used in generic function types.

**Comment by DavidBM** [comment]
Hey! 

I'm merging this and I will try to advance a bit more from here. We can always go back and leave the tests like there where before this pull request.

File: tests/unzip_file.rs [hunk 1]
```diff
@@ -1,65 +1,69 @@
...
-        remove_dir_all("tests/temp/zip_10MB").expect("The unzipped directory couldn't be deleted");
-    })
 }
 
 async fn hash_file(file: &str) -> Result<String, io::Error> {
     let mut file = File::open(file)?;
     let mut hasher = Sha256::new();
-    let _ = io::copy(&mut file, &mut hasher);
+    io::copy(&mut file, &mut hasher).unwrap();
+    drop(file);
```
Comment (elpiel) [inline tests/unzip_file.rs]: why do we have this `drop(file)`?

Comment (DavidBM) [reply tests/unzip_file.rs]: I was getting errors about the file not being deleted. For some time I pondered if the file was still opened, but not. At the end it was a different thing. I will delete it! Good that you told me

**Pull request merged**